R/zchunk_LA124.heat.R

```diff
@@ -69,117 +77,121 @@ module_energy_LA124.heat <- function(command, ...) {
+      mutate(value = if_else(is.na(value)|is.infinite(value), 0, value)) -> L124.heatoutratio_R_elec_F_tech_Yh
 
-    #Reset missing and infinite values (applicable for CC in the base years) to 0
-    L124.heatoutratio_R_elec_F_tech_Yh %>%
-      mutate(value=if_else(is.na(value)|is.infinite(value), 0, value))-> L124.heatoutratio_R_elec_F_tech_Yh
 
-    #Drop all rows where all years are 0
+    # Drop all rows where all years are 0
+    # Create a table of all the years of all 0
     L124.heatoutratio_R_elec_F_tech_Yh %>%
       group_by(GCAM_region_ID, technology, sector, fuel) %>%
-      summarise(sum=sum(value)) %>%
-      filter(sum==0) -> all_years_0
+      summarise(sum = sum(value)) %>%
+      filter(sum == 0) -> all_years_0
 
+    # Filter out the years in all_years_0 from heatoutratio
     L124.heatoutratio_R_elec_F_tech_Yh %>%
       left_join(all_years_0) %>%
-      mutate(sum=if_else(is.na(sum), 2, sum)) %>% #2 was a random, nonzero choice
-      filter(sum!=0)  %>%
+      #2 is a random, non-zero option for temporary use
+      mutate(sum = if_else(is.na(sum), 2, sum)) %>%
+      filter(sum != 0)  %>%
       select(-sum) -> L124.heatoutratio_R_elec_F_tech_Yh
 
     # This is complicated. Some historical regions / years have 0 district heat output (all fuels). This is fine unless there is heat output from
     # IEA's MAINCHP flow, which we model in the electric sector with a secondary output of heat to the district heat sector. In GCAM this will
     # cause solution failure as the district heat supply will be 0. In order to avoid this outcome we need a nominal amount of heat produced
     # in the zero years. The code below sets an exogenous year as the one to use to extract fuel shares for each region. In all 0 region / years,
     # output is equal to the output in the fuel-share year times 1e-3 to avoid throwing off the energy balances.
-    fuel_share_year <- "2010"
+
+    fuel_share_year <- 2010
     L124.in_EJ_R_heat_F_Yh %>%
       group_by(sector, GCAM_region_ID, year) %>%
-      summarise(value=sum(value)) %>%
-      mutate(value=if_else(value==0, 1e-3, value)) %>%
-      mutate(value=if_else(value!=1e-3, 0, value)) -> L124.mult_R_heat_Yh
+      summarise(value = sum(value)) %>%
+      # Creating an intermediary table - for years with 0 heat output, set value to 1e-3, for all others, set value to 0
+      mutate(value = if_else(value == 0, 1e-3, 0)) -> L124.mult_R_heat_Yh
 
     L124.out_EJ_R_heatfromelec_F_Yh %>%
       group_by(sector, GCAM_region_ID, year) %>%
-      summarise(elec_heat=sum(value)) %>%
-      mutate(elec_heat=if_else(elec_heat!=0, 1, elec_heat))-> L124.mult_R_heatfromelec_Yh
+      summarise(elec_heat = sum(value)) %>%
+      # Creating an intermediary table - for years with nonzero heat from CHP, set value to 1, for 0 values, leave as 0
+      mutate(elec_heat = if_else(elec_heat != 0, 1, elec_heat))-> L124.mult_R_heatfromelec_Yh
```

Comment:
Very confusing original code. I think 1 here is so that value * elect_heat below results in 1e-3. Isn't it clearer to set value = output_scaler (defined above) when elec_heat = 0 after line 157 instead.